test/site-builder-test.js

```diff
@@ -262,98 +259,94 @@ describe('SiteBuilder', function() {
+                'sender: mbland',
+                'cloning foo into ' + path.join(config.home, cloneDir),
+                'syncing to ' + config.s3.bucket + '/' +
+                  outputDir.replace(/\\/g, '/'),
+                'foo: build successful'
+              ],
+              expectedLog = expectedMessages.join('\n') + '\n';
+
+          expectLogMessages(logMsgs, expectedMessages);
+          expect(errorMsgs).to.be.empty;
+          expect(fs.readFileSync(path.join(config.home, buildLog), 'utf8'))
+            .to.equal(expectedLog);
+        })
+        .then(restoreLogs, restoreLogs);
     });
 
-    it('should create a builder that fails to build the site', function(done) {
-      var checkResult = check(done, function(err) {
-        var logMsgs = console.log.args,
-            errorMsgs = console.error.args,
-            expectedMessages = [
-              '18F/foo: starting build at commit deadbeef',
-              'description: Build me',
-              'timestamp: 2015-09-25',
-              'committer: michael.bland@gsa.gov',
-              'pusher: Mike Bland michael.bland@gsa.gov',
-              'sender: mbland',
-              'cloning foo into ' + cloneDir
-            ],
-            expectedErrors = [
-              'Error: failed to clone foo with exit code 1 from command: ' +
-                'git clone git@github.com:18F/foo.git --branch 18f-pages',
-              'foo: build failed'
-            ],
-            expectedLog = expectedMessages.concat(expectedErrors)
-              .join('\n') + '\n';
-
-        restoreLogs();
-        expect(err).to.be.null;
-        expectLogMessages(logMsgs, expectedMessages);
-        expectLogMessages(errorMsgs, expectedErrors);
-        expect(fs.readFileSync(buildLog, 'utf8')).to.equal(expectedLog);
-      });
+    it('should create a builder that fails to build the site', function() {
+      var handler = SiteBuilder.makeBuilderListener(webhook, builderConfig),
+          errMsg = 'Error: failed to clone foo ' +
+            'with exit code 1 from command: ' +
+            'git clone git@github.com:18F/foo.git --branch 18f-pages';
 
-      SiteBuilder.makeBuilderListener(webhook, builderConfig, checkResult);
-      var launcher = webhook.on.args[0][1];
       mySpawn.setDefault(mySpawn.simple(1));
       captureLogs();
-      launcher(incomingPayload);
+      return handler(incomingPayload).should.be.rejectedWith(errMsg)
+        .then(function() {
+          var logMsgs = console.log.args,
+              errorMsgs = console.error.args,
+              expectedMessages = [
+                '18F/foo: starting build at commit deadbeef',
+                'description: Build me',
+                'timestamp: 2015-09-25',
+                'committer: michael.bland@gsa.gov',
+                'pusher: Mike Bland michael.bland@gsa.gov',
+                'sender: mbland',
+                'cloning foo into ' + path.join(config.home, cloneDir)
+              ],
```

Comment:
Actually, all these messages are generated by the pages-server, not by Git or anything else.